test/unit/data_set_test.rb

```diff
@@ -395,5 +395,19 @@ class DataSetTest < ActiveSupport::TestCase
       end
     end
 
+    context "for a valid postcode with no places found" do
+      setup do
+        @service = FactoryGirl.create(:service)
+        @data_set = @service.latest_data_set
+      end
+
+      should "return an empty array for valid postcode" do
+        stub_mapit_postcode_response_from_fixture("JE4 5TP")
+
+        areas = @data_set.places_for_postcode("JE4 5TP").map(&:areas)
```

Comment:
I don't think you need the `map` call here - `places_for_postcode` returns an array of places, and places don't have areas.